### DISCUSSION TITLE
Is it possible to include multiple API's scenario and config details in single yaml file?

### DISCUSSION THREAD

**Discussion by rrnareshkumar** [artilleryio/artillery#2112] [Artillery]
Hi , I would like to load test multiple API's at the same time . May i know if it is possible to include multiple API's scenario and config details in single(same) yaml file or do we need to create separate yaml file for each API ? Kindly suggest

**Comment by bernardobridge** [comment]
Hi @rrnareshkumar .

It depends a little bit on what you're trying to do with it, but the general answer is _**yes you can**_:

**For scenarios:**
You can specify multiple scenarios and then use the `weights feature` (documented here: https://www.artillery.io/docs/reference/test-script#scenario-weights). Those scenarios will then be picked up by virtual users probabilistically, which helps simulate real application usage (i.e. you'd have multiple users arriving at your application and going through different scenarios which may be more or less likely).

**For configs:**
You can leverage `environments` (documented here: https://www.artillery.io/docs/reference/test-script#environments---config-profiles) in your config to keep multiple config types for different purposes.

**Comment by rrnareshkumar** [comment]
Hi , thanks for sharing the links . 

I did refer the above link , environments config profiles and tried it (posted sample code below) . I kept  testapp1.com in "production " environment and i kept testapp2.com in "local" environment and i was able to run test using production or local environment successfully but my requirement is to have test 2 different application API's (example : testapp1.com and testapp2.com) in same config file and execute it together (ex : may be in other terms , i  want to run both production and local environment  ) 

May i know if it is possible ? please suggest the options and steps to test 2 different application API's together .



 environments:
    production:
      target: "http://testapp1.com"
      phases:
        - duration: 1200
          arrivalRate: 10
    local:
      target: "http://testapp2.com"
      phases:
        - duration: 1200
          arrivalRate: 20

**Comment by bernardobridge** [comment]
Hello 👋. If you want to test more than one target in the same scenario file, you should be able to do that by overriding the `url` inside flow with a full URL. For example:

```yaml
config:
  target: "somethingdoesntmatter"
  phases:
    - duration: 1200
      arrivalRate: 20
      name: "Phase 1"
scenarios:
  - name: Scenario 1
    weight: 3
    flow:
      - get:
          url: "http://testapp1.com"
  - name: Scenario 2
    weight: 1
    flow:
      - get:
          url: "http://testapp2.com"
```

The above will pick Scenario 1 approximately every 3 out of 4 times (75% of the times), which will use `testapp1`, and Scenario 2 25% of the times (which will use `testapp2`). 

---

_Note: Artillery is a very versatile tool so there may be multiple ways to do things. But as you can understand, we don't know the internal of your applications and use case, and won't be able to help you customise everything as well as you can. If the above is still not quite what you wanted, I suggest you spend some time reading the documentation at https://www.artillery.io/docs to understand how Artillery works better, as you'll be in the best position to use it for your needs. We are happy to take improvement feedback if you then find gaps. Thank you!_
